Jenkinsfile

```
@@ -1,153 +1,103 @@
 #!/usr/bin/env groovy
 
-def dockerRegistry = "329802642264.dkr.ecr.eu-west-1.amazonaws.com"
-def nodeImageVersion = "10.16.0"
-def nodeImage = "${dockerRegistry}/bbc-news/node-10-lts:${nodeImageVersion}"
+def nodeImage = "329802642264.dkr.ecr.eu-west-1.amazonaws.com/bbc-news/node-10-lts:10.16.0"
 
-def nodeName
 def slackChannel = "#si_repo-psammead"
-def stageName = "Unknown"
-def gitCommitAuthor = "Unknown"
-def gitCommitMessage = "Unknown"
 
-def getCommitInfo = {
-  gitCommitAuthor = sh(returnStdout: true, script: "git --no-pager show -s --format='%an' ${GIT_COMMIT}").trim()
-  gitCommitMessage = sh(returnStdout: true, script: "git log -1 --pretty=%B").trim()
+def notifySlack(colour, buildStatus, gitCommitAuthor, info, gitCommit, gitCommitMessage, slackChannel) {
+  // call the global slackSend method in Jenkins
+  slackSend(color: colour, message: "*${buildStatus}* on ${env.BRANCH_NAME} [build ${BUILD_DISPLAY_NAME}] \n*Author:* ${gitCommitAuthor} \n*Info:* ${info} \n*Commit Hash* \n${gitCommit} \n*Commit Message* \n${gitCommitMessage}", channel: slackChannel)
 }
 
-def notifySlack(colour, buildStatus, gitCommitAuthor, stageName, gitCommitMessage, slackChannel) {
-  // call the global slackSend method in Jenkins
-  slackSend(color: colour, message: "*${buildStatus}* on ${GIT_BRANCH} [build ${BUILD_DISPLAY_NAME}] \n*Author:* ${gitCommitAuthor} \n*Stage:* ${stageName} \n*Commit Hash* \n${GIT_COMMIT} \n*Commit Message* \n${gitCommitMessage}", channel: slackChannel)
+def cleanUp() {
+  sh 'chmod -R 777 .git'
 }
 
-pipeline {
-  agent any
-  options {
-    timeout(time: 60, unit: 'MINUTES')
-    timestamps ()
-  }
-  environment {
-    CI = true
-    CC_TEST_REPORTER_ID = '06c1254d7c2ff48f763492791337193c8345ca8740c34263d68adcc449aff732'
-  }
-  stages {
-    stage ('Run application tests') {
-      agent {
-        docker {
-          image "${nodeImage}"
-          label nodeName
-          args '-u root -v /etc/pki:/certs'
-        }
-      }
-      steps {
-        sh 'ls' // quick debug step
-        sh 'rm -rf ./app'
-        script {
-          if (GIT_BRANCH == 'latest') {
-            getCommitInfo()
+node {
+  timeout(time: 30, unit: 'MINUTES') {
+    withEnv([
+      'CI=true',
+      'CC_TEST_REPORTER_ID=06c1254d7c2ff48f763492791337193c8345ca8740c34263d68adcc449aff732'
+    ]) {
+      cleanWs()
+
+      // git checkout
+      checkout scm
+
+      // get git commit info for notifications
+      gitCommitHash = sh(returnStdout: true, script: "git log -n 1 --pretty=format:'%h'").trim()
+      gitCommitAuthor = sh(returnStdout: true, script: "git --no-pager show -s --format='%an' ${gitCommitHash}").trim()
```

nit picking - why not have the shell commands for gitCommitAuthor and gitCommitMessage match? They're doing roughly the same operation
```suggestion
      gitCommitAuthor = sh(returnStdout: true, script: "git log -1 --pretty=%an").trim()
```